configuration/exodus-pipeline.yaml

```diff
@@ -120,39 +120,35 @@ Resources:
       RoleArn: !Sub arn:aws:iam::${AWS::AccountId}:role/${project}-codepipe-${env}
       Stages:
         # Create Github source for dev, stage pipelines
-        - !If
-          - NotProd
-          - Name: Source
-            Actions:
-              - Name: Source
-                Namespace: SourceVars
-                ActionTypeId:
-                  Category: Source
-                  Owner: ThirdParty
-                  Provider: GitHub
-                  Version: 1
-                Configuration:
-                  Owner: !Ref repoOwner
-                  Repo: !Ref repoName
-                  Branch:
-                    !If
-                      - NoBranch
-                      - !If
-                        - IsDev
-                        - master
-                        - deploy
-                      - !Ref repoBranch
-                  OAuthToken: !Ref githubToken
-                  PollForSourceChanges: false
-                OutputArtifacts:
-                  - Name: SourceArtifact
-          - !Ref AWS::NoValue
-        # Create S3 source for prod pipeline
-        - !If
-          - IsProd
-          - Name: Source
-            Actions:
-              - Name: Source
+        - Name: Source
+          Actions:
+            - Name: Source
```

Comment:
Doesn't this mean that prod pipeline will now be triggered on pushes to the deploy branch as well as updates to the template by the stage pipeline?

At least the stage pipeline is still stopped if the commit isn't verified. With prod, any time anything is pushed to the deploy branch, it'll go through an unnecessary redeployment. Prod will either run once if the push was illegitimate or twice if it was (one for the push [ineffective], one for the template update).

I don't know the full financial impact of this but it's money wasted. Anything we can do to avoid that, if so?

Reply:
Good point, but the git webhook is only configured for dev and stage envs, so I don't think this should be a problem. My [prod pipeline history](https://console.aws.amazon.com/codesuite/codepipeline/pipelines/caleigh-pipeline-prod/executions?region=us-east-1) only contains executions from the trigger `CloudWatchEvent - arn:aws:events:us-east-1:333296168829:rule/caleigh-pipeline-prod-CodePipelineEventRule-2WXYIRNUXELZ`; as opposed to my [stage pipeline history](https://console.aws.amazon.com/codesuite/codepipeline/pipelines/caleigh-pipeline-stage/executions?region=us-east-1), which only contains executions from the trigger `Webhook - arn:aws:codepipeline:us-east-1:333296168829:webhook:caleigh-pipeline-stage`.

Reply:
Glad to be wrong. Thanks!